### PR TITLE
backupccl: add telemetry for deprecated forms of backupccl cmds

### PR DESCRIPTION
**Pull request by msbutler**
Release note: None

### CODE REVIEW

File: pkg/ccl/backupccl/restore_planning.go [hunk 1]
```diff
@@ -1795,6 +1795,11 @@ func doRestorePlan(
 		if restoreStmt.DescriptorCoverage == tree.AllDescriptors {
 			telemetry.Count("restore.full-cluster")
 		}
+		if restoreStmt.Subdir == nil {
+			telemetry.Count("restore.deprecated-syntax")
```
Comment (rhu713) [inline pkg/ccl/backupccl/restore_planning.go:1799]: should this be something like `deprecated-subdir-syntax` to give it more description?

File: pkg/ccl/backupccl/show.go [hunk 2]
```diff
@@ -448,7 +449,15 @@ you must pass the 'encryption_info_dir' parameter that points to the directory o
 						len(dest)))
 			}
 		}
-		return infoReader.showBackup(ctx, &mem, mkStore, info, p.User(), resultsCh)
+		if err := infoReader.showBackup(ctx, &mem, mkStore, info, p.User(), resultsCh); err != nil {
+			return err
+		}
+		if backup.InCollection == nil {
+			telemetry.Count("show-backup.deprecated_syntax")
```
Comment (rhu713) [inline pkg/ccl/backupccl/show.go:456]: ditto on adding more description to the metric.

File: pkg/ccl/backupccl/backup_planning.go [hunk 3]
```diff
@@ -975,9 +976,30 @@ func collectTelemetry(
 	}
 	if backupDetails.CollectionURI != "" {
 		countSource("backup.nested")
-		if initialDetails.Destination.Subdir == latestFileName {
-			countSource("backup.into-latest")
```
[the remark this answers is not in the document]
Comment (msbutler) [reply pkg/ccl/backupccl/backup_planning.go:979]: Resolved offline that this is ok

File: pkg/ccl/backupccl/restore_planning.go (hunk 1 above)
Comment (msbutler) [reply pkg/ccl/backupccl/restore_planning.go]: Done.

File: pkg/ccl/backupccl/show.go (hunk 2 above)
Comment (msbutler) [reply pkg/ccl/backupccl/show.go]: Done.

**Review by rhu713 — APPROVED** [review]

**Comment by msbutler** [comment]
TFTR!

bors r = rhu713

**Comment by msbutler** [comment]
bors r=rhu713

**Comment by msbutler** [comment]
unrelated flake

bors r=rhu713

**Pull request merged**